### PR TITLE
Fix invalid ID in file upload request

### PR DESCRIPTION
**Pull request by krdabrowski**

### CODE REVIEW

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/TransformResponse.java [hunk 1]
```diff
@@ -23,22 +23,19 @@
...
-@NoArgsConstructor
-@AllArgsConstructor
-public class TransformResponse
-{
-    @NotBlank
-    private String targetReference;
-}
+public record TransformResponse(
+        @NotBlank String targetReference,
+        @NotNull @JsonDeserialize(using = ClientDataDeserializer.class) ClientData clientData)
```
Comment (Pawel-608) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/TransformResponse.java:40]: I think you can add this deserializer in `MappingConfig` instead

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/model/ClientDataDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.model;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.jackson.RawJsonDeserializer;
+
+public class ClientDataDeserializer extends RawJsonDeserializer<ClientData>
```
Comment (Pawel-608) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/model/ClientDataDeserializer.java:30]: I would move this class to config/jacson package tbh

**Review by Pawel-608 — APPROVED** [review]
LGTM, just small remarks regarding jackson stuff

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/TransformResponse.java (hunk 1 above)
Comment (krdabrowski) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/TransformResponse.java:40]: Annotations are more descriptive IMO and tell straightforwardly that something is handled differently, in the other hand I don't mind hiding this in `MappingConfig` and simplifying record 🤔

**Review by tpage-alfresco — APPROVED** [review]
LGTM

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/TransformResponse.java (hunk 1 above)
Comment (krdabrowski) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/TransformResponse.java:40]: It turned out, that when I register the deserilizer globally then the `RawJsonDeserializer` which uses global `ObjectMapper` calls itself again in some kind of circular relation but without the JSON value and fails. I will leave it as it is and we can change it in the future.

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/model/ClientDataDeserializer.java (hunk 2 above)
Comment (krdabrowski) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/model/ClientDataDeserializer.java]: Done

**Pull request merged**